### PR TITLE
Updates the building a plugin guide

### PR DESCRIPTION
**Pull request by josh146**
**Description of the Change:**

Updates the building a plugin guide to:

* Fix broken links

* Change the structure slightly (separate the 'Applying operations' section into 'Applying Operations', 'Measuring observables', and 'Advanced execution').

* Add info on tensor observable API

**Benefits:** links are fixed, now back up to date.

**Possible Drawbacks:** n/a

**Related GitHub Issues:** #362

### CODE REVIEW

File: doc/development/plugins.rst [hunk 1]
```diff
@@ -43,54 +58,67 @@ This is as simple as importing the abstract base class :class:`~.Device` from Pe
...
-  This is used to decide whether an operation is supported by your device in the default implementation of the public method :meth:`~.Device.supports_operation`.
+  This is used to decide whether an operation is supported by your device in the
+  default implementation of the public method :meth:`.Device.supports_operation`.
 
-* :attr:`~.Device.observables`: set of the supported PennyLane observables as strings, e.g.,
+* :attr:`.Device.observables`: set of the supported PennyLane observables as strings, e.g.,
 
   .. code-block:: python
 
     observables = {"QuadOperator", "NumberOperator", "X", "P"}
```
Comment (johannesjmeyer) [inline doc/development/plugins.rst]: Maybe one should not mix qubit operations and cv observables in the explanation? That could confuse people.

File: doc/development/plugins.rst [hunk 2]
```diff
@@ -180,23 +293,39 @@ PennyLane uses a ``setuptools`` ``entry_points`` approach to plugin discovery/in
...
-^^^^^^^^^^^^^^^^^^^^^^^^^
+-------------------------
 
-If you would like to support an operation or observable that is not currently supported by PennyLane, you can subclass the :class:`~.Operation` and :class:`~.Observable` classes, and define the number of parameters the operation takes, and the number of wires the operation acts on. For example, to define the Ising gate :math:`XX_\phi` depending on parameter :math:`\phi`,
+If you would like to support an operation or observable that is not currently supported by
+PennyLane, you can subclass the :class:`~.Operation` and :class:`~.Observable` classes, and
+define the number of parameters the operation takes, and the number of wires the operation
+acts on. For example, to define the Ising gate :math:`XX_\phi` depending on parameter :math:`\phi`,
 
 .. code-block:: python
```
Comment (johannesjmeyer) [inline doc/development/plugins.rst]: I don't know about the Ising gate -- does it really have the default gradient recipe?

Comment (josh146) [reply doc/development/plugins.rst]: Good point - I guess I'll need to go through the maths and see.

Maybe I should just change the name to `CustomOperation` :laughing:

Comment (johannesjmeyer) [reply doc/development/plugins.rst]: I think that is actually a good idea :+1:

**Pull request merged**